deeppavlov/dataset_readers/ubuntu_dstc7_mt_reader.py

```diff
@@ -0,0 +1,140 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Tuple, Dict
+from pathlib import Path
+import json
+import numpy as np
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+
+
+@register('ubuntu_dstc7_mt_reader')
+class UbuntuDSTC7MTReader(DatasetReader):
+    """
+    DatasetReader for Ubuntu Dialogue Corpus Dataset (version 3), prepared for DSTC 7 competition Track 1 Subtrack 1.
+
+    https://github.com/IBM/dstc7-noesis
+
+    Args:
+        data_path (str): A path to a folder with dataset json files.
+        num_context_turns (int): A maximum number of dialogue ``context`` turns.
+        num_responses (int): A number of responses for each context; default is equal to all 100 responses,
+            it can be reduced to 10 (1 true response + 9 random wrong responses) to adapt with succeeding pipeline
+        padding (str): "post" or "pre" context sentences padding
+    """
+    
+    def read(self,
+             data_path: str,
+             num_context_turns: int = 10,
+             num_responses: int = 100,
+             padding: str = "post",
+             *args, **kwargs) -> Dict[str, List[Tuple[List[str], int]]]:
+
+        self.num_turns = num_context_turns
+        self.padding = padding
+        self.num_responses = num_responses
+
+        dataset = {}
+        dataset["train"] = self._create_dialog_iter(Path(data_path) / 'ubuntu_train_subtask_1.json', "train")
+        dataset["valid"] = self._create_dialog_iter(Path(data_path) / 'ubuntu_dev_subtask_1.json', "valid")
+        dataset["test"]  = self._create_dialog_iter(Path(data_path) / 'ubuntu_test_subtask_1.json', "test")
+        return dataset
+
+    def _create_dialog_iter(self, filename, mode="train"):
+        """
+        Read input json file with test data and transform it to the following format:
+        [
+            ( [context_utt_1, ..., context_utt_10, response_utt_1, ..., response_utt_N], label ),
+            ( [context_utt_1, ..., context_utt_10, response_utt_1, ..., response_utt_N], label ),
+            ...
+        ]
+
+        where
+        * [context_utt_1, ..., context_utt_10, response_utt_1, ..., response_utt_N] - list that consists of
+        ``num_context_turn`` utterances, followed by ``num_responses`` responses.
+        Where
+        * label - label of the sample
+
+        Args:
+            filename (Path): filename to read
+            mode (str): which dataset to return. Can be "train", "valid" or "test"
+
+        Returns:
+             list of contexts and responses with their labels. More details about the format are provided above
+        """
+        data = []
+        with open(filename, 'rb') as f:
+            json_data = json.load(open(filename, 'rt'))
+            for entry in json_data:
```

Comment:
```python
with open(filename, encoding='utf8') as f:
    json_data = json.load(f)

for entry in json_data:
    ...
```
When `open` not as bytes, encoding is necessary because on different systems it defaults to different values.